File: Components/eXpressApp Framework/EFDemoCodeFirst/CS/EFDemo.Module/Data/Department.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

using DevExpress.Persistent.Base;

namespace EFDemo.Module.Data {
    [DefaultClassOptions]
    [DefaultProperty("Title")]
    public class Department {
		public Department() {
			Positions = new List<Position>();
			Contacts = new List<Contact>();
		}
        //[Browsable(false)]
        [VisibleInDetailView(false), VisibleInListView(false), VisibleInLookupListView(false)]
		public Int32 ID { get; protected set; }
		public String Title { get; set; }
		public String Office { get; set; }
		public virtual IList<Position> Positions { get; set; }
		public virtual IList<Contact> Contacts { get; set; }
	}
}

File: Components/eXpressApp Framework/MainDemo/VB/MainDemo.Module/BusinessObjects/Department.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace MainDemo.Module.BusinessObjects
    <DefaultClassOptions(), System.ComponentModel.DefaultProperty("Title")> _
    Public Class Department
        Inherits BaseObject
        Private title_Renamed As String
        Private office_Renamed As String
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Title() As String
            Get
                Return title_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Title", title_Renamed, value)
            End Set
        End Property
        Public Property Office() As String
            Get
                Return office_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Office", office_Renamed, value)
            End Set
        End Property
        <Association("Department-Contacts")> _
        Public ReadOnly Property Contacts() As XPCollection(Of Contact)
            Get
                Return GetCollection(Of Contact)("Contacts")
            End Get
        End Property
        <Association("Departments-Positions")> _
        Public ReadOnly Property Positions() As XPCollection(Of Position)
            Get
                Return GetCollection(Of Position)("Positions")
            End Get
        End Property
    End Class
End Namespace
